http3/headers.go

```diff
@@ -178,25 +178,24 @@ func hostnameFromURL(url *url.URL) string {
 	return ""
 }
 
-func responseFromHeaders(headerFields []qpack.HeaderField) (*http.Response, error) {
+func updateResponseFromHeaders(rsp *http.Response, headerFields []qpack.HeaderField) error {
```

Comment:
I suggest adding some documentation here, since I was at first surprised by the signature of this function:
```suggestion
// updateResponseFromHeaders sets up http.Response as an HTTP/3 response,
// using the decoded qpack header filed.
// It is only called for the HTTP header (and not the HTTP trailer).
// It takes an http.Response as an argument to allow the caller to set the trailer later on.
func updateResponseFromHeaders(rsp *http.Response, headerFields []qpack.HeaderField) error {
```